providers/azure.go

```diff
@@ -360,6 +367,30 @@ func (p *AzureProvider) redeemRefreshToken(ctx context.Context, s *sessions.Sess
 	return nil
 }
 
+func (p *AzureProvider) getUser(ctx context.Context, accessToken string) (string, error) {
+	header := makeAzureHeader(accessToken)
+	endpoint := p.Data().ValidateURL.String()
+
+	var jsonResponse struct {
+		Name string `json:"name"`
+	}
+
+	err := requests.New(endpoint).
+		WithContext(ctx).
+		WithHeaders(header).
+		Do().
+		UnmarshalInto(&jsonResponse)
```

Comment:
I'm concerned about how many requests we make individually to the profile API. I'm sure I was working on a claim extractor tool that would make sure we call the profile API at most once per session creation. @tuunit we should make sure to get that project across the finish line

Reply:
I think I did finish it, `util.NewClaimExtractor`, why is this provider not using the claim extractor? 🤔